### ISSUE TITLE
[BUG][typescript-inversify] Type mismatch between service model and HttpClient requirement

### ISSUE THREAD

**Issue by siada** [OpenAPITools/openapi-generator#3618]
#### Bug Report Checklist

- [x] Have you provided a full/minimal spec to reproduce the issue?
- [x] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [x] What's the version of OpenAPI Generator used?
- [x] Have you search for related issues/PRs?
- [ ] What's the actual output vs expected output?
- [ ] [Optional] Bounty to sponsor the fix ([example](https://www.bountysource.com/issues/66123212-javascript-client-produces-a-wrong-object-for-a-string-enum-type-that-is-used-with-ref))

##### Description

When using typescript in `strict` mode, the code won't compile if your spec has an optional requestBody because the HttpClient defines `body` as `{}` which means 'any except null or undefined'

##### openapi-generator version

4.1.0

##### Steps to reproduce

Generated service method:

```typescript
    public apiAccountLoginPost(loginViewModel?: LoginViewModel, observe?: 'body', headers?: Headers): Promise<LoginViewModelResult>;
    public apiAccountLoginPost(loginViewModel?: LoginViewModel, observe?: 'response', headers?: Headers): Promise<HttpResponse<LoginViewModelResult>>;
    public apiAccountLoginPost(loginViewModel?: LoginViewModel, observe: any = 'body', headers: Headers = {}): Promise<any> {
```

HttpClient:

```typescript
    post(url: string, body: {}|FormData, headers?: Headers): Observable<HttpResponse> {
        return this.performNetworkCall(url, "POST", this.getJsonBody(body), this.addJsonHeaders(headers));
    }
```

Error: 

![image](https://user-images.githubusercontent.com/177663/62855211-810c9c80-bce9-11e9-9b34-5a7c4a477405.png)

##### Related issues/PRs

#3607

##### Suggest a fix

I believe changing the `{}` to `any` in the HttpClient will resolve this, however I don't know enough about typescript to understand the impact this may have

I'm happy to organise the PR for this if you agree with this change

**Comment by macjohnny** [comment]
how about changing it to `body?: Object|FormData` and also adapting 
https://github.com/siada/openapi-generator/blob/b54a5266ac9824306e5904554275d809dcd66057/modules/openapi-generator/src/main/resources/typescript-inversify/HttpClient.mustache#L40
to handle undefined/null values?

**Comment by macjohnny** [comment]
@gualtierim what is your opinion?

**Comment by siada** [comment]
For handling the undefined/null values should getJsonBody return null, empty string, or `{}` ?

**Comment by macjohnny** [comment]
> For handling the undefined/null values should getJsonBody return null, empty string, or `{}` ?

good question, I guess `undefined` ?

**Comment by gualtierim** [comment]
I agree to change the signature of httpClient with Object|FormData, getJsonBody should return undefined but I will try the next week

**Comment by deluksic** [comment]
Also, `addJsonHeaders`'s `headers` parameter should be optional.

**Comment by siada** [comment]
@deluksic I already fixed that in PR #3607 this issue has been opened as a discussion from that

**Comment by siada** [comment]
@gualtierim Did you manage to get around to trying the solution out yet?
